src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -25,6 +25,9 @@ public class ViewBuffer : IHtmlContentBuilder
         private readonly IViewBufferScope _bufferScope;
         private readonly string _name;
         private readonly int _pageSize;
+        private ViewBufferPage _singlePage = null;          // Limits allocation if the ViewBuffer has only one page (frequent case).
```

Comment:
Gotcha. Maybe split it out into a separate scenario? It wasn't entirely obvious that you were trying to test use after Clear there. 

Test1 - Append_CreatesManyPages 
Test2 - Append_WorksAfterPagesHasBeenCleared (or something like that)

Makes it easier to figure out what specific scenario is broken if we happen to change this in the future.